### ISSUE TITLE
Support for asyncio

### ISSUE THREAD

**Issue by zhulik** [femueller/python-n26#105]
Hello,

Thank you for the amazing package!

I'm trying to ressurect the home assistant integration mention in your readme, and found out that the package is written in a synchronous way, which makes it much harder to properly use it in the home assistant environment which is fully async.

So I just wanted to ask if there are any plans to migrate the project to asyncio. I believe it's possible to migrate it and keep the synchronous API in place as a backward compatible wrapper around asynchronous code, what do you think?

**Comment by femueller** [comment]
Hey @zhulik,

Thanks for your nice words :)

There were currently no plans to support asyncio for `python-n26` and I don't know if it solves the current problem to support a proper integration for Home Assistant.
As far as I know in the beginning @markusressel built support for Home Assistant before N26 introduced the PSD2 authentication.
Currently, you need to refresh the data by hand using your second factor (N26 paired smartphone) to update HA... which is definitely not the ideal integration.

I heard rumors that N26 will provide developer tokens in the future (this or next year), so that integrations like this can be built and used properly.

**Comment by zhulik** [comment]
I've already tried to fix the integration using the last version of the package, and it works. I only needed to use my phone during the first startup after the integration was added. The tokens were stored in a file, so it should survive restarts and upgrades. The only problem is that I need to wrap all calls to python-n26 as it's described in [this post](https://www.roguelynn.com/words/asyncio-sync-and-threaded/) to make them play well with the event loop.

**Comment by markusressel** [comment]
No, it won't work. Authentication tokens are only valid for 2 hours max, period. Currently, there is no way around that.
See also this discussion I had about this not too long ago: https://github.com/femueller/python-n26/discussions/101

I really appreciate your interest and effort though! I would be _very_ happy if there was a way around this, but afaik there is none. We can definitely have a chat about supporting asyncio though, as I think it would be a valuable addition to the project. I can already say that I won't implement it myself, but I am happy to review PRs 👍 🤓
